src/phonetic_manager.c

```diff
@@ -22,6 +24,10 @@ static void PhoneticManager_AddPrefix(char** phoneticTerm) {
 
 void PhoneticManager_ExpandPhonetics(PhoneticManagerCtx* ctx, const char* term, size_t len,
                                      char** primary, char** secondary) {
+  if(len > MAX_TERM_TO_EXPAND_SIZE) {
```

Comment:
Should we fallback to heap based buffer in this case?

Reply:
Yes, you are right. Please review the last change. Thanks.